spectec/spec/wasm-1.0/8-reduction.watsup

```diff
@@ -15,6 +15,24 @@ rule Step/read:
   -- Step_read: z; instr* ~> instr'*
 
 
+;; Expressions
+
+relation Eval: config ~>* state; val*  hint(show "E-expr")
+relation Eval_expr: state; expr ~>* state; val*  hint(show "E-expr")
+
+rule Eval/done:
+  z; val* ~>* z; val*
+
+rule Eval/step:
+  z; admininstr*  ~>*  z''; val*
+  -- Step: z; admininstr*  ~>  z'; admininstr'*
+  -- Eval: z'; admininstr'  ~>*  z''; val*
```

Comment:
Declaratively, it would be more natural to have this more general relation (while leaving Eval_expr as is except invoking Steps):
```suggestion
relation Steps: config ~>* config  hint(show "E")

rule Steps/refl:
  z; admininstr* ~>* z; admininstr*

rule Steps/trans:
  z; admininstr*  ~>*  z''; admininstr''*
  -- Step: z; admininstr*  ~>  z'; admininstr'*
  -- Steps: z'; admininstr'  ~>*  z''; admininstr''*
```
Would that work in the backend as well?

Reply:
Yes, it should work well without any problem.

(It was already handled in hardcoded way. The rule premise `Eval_expr: lhs ~> rhs` is rougly translated to the AL statement `1. Let rhs be the result of computing $eval_expr(lhs)`, where `eval_expr` is a manually written algorithm)